main.js

```diff
@@ -0,0 +1,253 @@
+    const BASE_URL = `https://wttr.in/${pickedLocation}?format=j1`
+    // Fetching BASE_URL
+    fetch(BASE_URL)
+        .then((response) => response.json())
+        // Using destructuring to unpack values from fetch response
+        .then(({nearest_area, current_condition, weather}) => {
+            // Creating variable to store value of current temperature 
+            const currentTempF = current_condition[0].temp_F;
+            // Creating variable to store value of nearest_area[0].region[0].value
+            const regionValue = nearest_area[0].region[0].value;
+            // Creating variable to store value of nearest_area[0].country[0].value
+            const country = nearest_area[0].country[0].value;
+            // Creating variable to store value for currently feels like temp in F 
+            const currentlyFeelsF = current_condition[0].FeelsLikeF;
+            // Creating variable to store value for today's average temp in F
+            const todayAvgTempF = weather[0].avgtempF;
+            // Creating variable to store value for today's max temp in F
+            const todayMaxTempF = weather[0].maxtempF;
+            // Creating variable to store value for today's min temp in F
+            const todayMinTempF = weather[0].mintempF;
+            // Creating variable to store value for tomorrow's average temp in F
+            const tomorrowAvgTempF = weather[1].avgtempF;
+            // Creating variable to store value for tomorrow's max temp in F
+            const tomorrowMaxTempF = weather[1].maxtempF;
+            // Creating variable to store value for tomorrow's min temp in F
+            const tomorrowMinTempF = weather[1].mintempF;
+            // Creating  variable to store value for day after tomorrow's average temp in F
+            const dayAfterTomorrowAvgTempF = weather[2].avgtempF;
+            // Creating variable to store value for day after tomorrow's max temp in F
+            const dayAfterTomorrowMaxTempF = weather[2].maxtempF;
+            // Creating variable to store value for day after tomorrow's min temp in F
+            const dayAfterTomorrowMinTempF = weather[2].mintempF;
+            // Creating variable to store value for nearest area
+            const nearestArea = nearest_area[0].areaName[0].value;
+            // Creating variable to store value for today's chance of sunshine
+            const chanceOfSunshine = weather[0].hourly[0].chanceofsunshine;
+            // Creating variable to store value for today's chance of rain
+            const chanceOfRain = weather[0].hourly[0].chanceofrain;
+            // Creating variable to store value for today's chance of snow
+            const chanceOfSnow = weather[0].hourly[0].chanceofsnow;
+
+
+            // Calling functions to pass cypress tests
+            getDetailsFromTextInput();
+            displayRegion(regionValue);
+            displayCountry(country);
+            displayCurrentlyFeelsF(currentlyFeelsF);
```

Comment:
the displayRegion, displayCountry, and similar functions can be one function. There isn't any benefit from having them separate when their functionality is so similar